beta/src/pages/apis/react/Suspense.md

```diff
@@ -2,17 +2,126 @@
+- Once <CodeStep step={2}>`<Post />`</CodeStep> finishes loading, <CodeStep step={1}>`<BigSpinner />`</CodeStep> is replaced by the main content.
+- If <CodeStep step={4}>`<Comments />`</CodeStep> hasn't loaded yet, <CodeStep step={3}>`<CommentsGlimmer />`</CodeStep> is shown in its place.
+- Finally, once <CodeStep step={4}>`<Comments />`</CodeStep> finishes loading, it replaces <CodeStep step={3}>`<CommentsGlimmer />`</CodeStep>. 
+
+### Lazy loading components with Suspense {/*suspense-for-code-splitting*/}
+
+The [`lazy`](/apis/react/lazy) API is powered by Suspense. When you render a component imported with `lazy`, it will suspend if it hasn't loaded yet.
+
+```js [[1, 10, "<LoadingSpinner />"], [2, 11, "<MoreInfo />"]]
+import { Suspense, lazy } from 'react';
+
+const MoreInfo = lazy(() => import('./MoreInfo'));
+
+function Details({showMore}) {
+  return (
+    <>
+      <Info />
+      {showMore ? (
+        <Suspense fallback={<LoadingSpinner />}>
+          <MoreInfo />
+        </Suspense>
+      ) : null}
+    </>
+  );
+}
+```
+
+In this example, the code for <CodeStep step={2}>`<MoreInfo />`</CodeStep> won't be loaded until you attempt to render it. If <CodeStep step={2}>`<MoreInfo />`</CodeStep> hasn't loaded yet, <CodeStep step={1}>`<LoadingSpinner />`</CodeStep> will be shown in its place.
+
+### Fetching data with Suspense {/*suspense-for-data-fetching*/}
+
+<Note>
+The requirements for implementing a Suspense-enabled data source are currently unstable and undocumented. An official API for integrating data sources with Suspense is in progress, and will be released soon in an upcoming minor.
+</Note>
+
+General support for data fetching with Suspense without the use of a framework is not yet officially supported. However, you can use Suspense for data fetching in opinionated frameworks like [Relay](https://relay.dev/docs/guided-tour/rendering/loading-states/), [Next.js](https://nextjs.org/docs/advanced-features/react-18), [Hydrogen](https://hydrogen.shopify.dev/), and [Remix](https://remix.run/).
+
+## Reference {/*reference*/}
+
+### Props {/*props*/}
+* `children`: The actual UI you intend to render. If `children` suspends while rendering, the Suspense boundary will switch to rendering `fallback`.
+* `fallback`: An alternate UI to render in place of the actual UI if it has not finished loading. Any valid React node is accepted, though in practice, a fallback is a lightweight placeholder view, such as a loading spinner or skeleton. Suspense will automatically switch to `fallback` when `children` suspends, and back to `children` when the data is ready. If `fallback` suspends while rendering, it will activate the next parent Suspense boundary.
+
+## Troubleshooting {/*troubleshooting*/}
+
+### How do I prevent the UI from being replaced by a fallback during an update? {/*preventing-unwanted-fallbacks*/}
```

Comment:
Mention useTransition for a loading indicator? Folks may think they don't want startTransition because there's no visual feedback.

Reply:
Yeah I can mention `useTransition`. Though I don't want to get into a full explanation of transitions here. I believe the plan is to have a comprehensive guide to transitions in the "Learn" section similar to what we have for Effects. When I chatted to @gaearon about it, the idea was to sort of assume that doc already exists when writing this API doc. The other reason I didn't mention `isPending` is that a lack of a loading indicator is still preferable to UI disappearing.

Reply:
I don't think a full explanation is necessary but I think mentioning how to solve a problem is useful. Like, assume that the doc exists so you can refer to the concept.

Reply:
I guess I just disagree it’s relevant to this particular troubleshooting section (how do I prevent UI from switching to a fallback?). Seems more appropriate for the useTransition or startTransition docs, which aren’t written yet but this section can link to.

Reply:
Maybe in a different section like “How do I show loading indicator while a transition is suspended?” that can link out to the useTransition docs.

Reply:
Well, I'm thinking it's a part of *using Suspense*. Maybe this shouldn't go in Troubleshooting but in Usage instead. What I'm trying to do with Usage is to surface related APIs that solve problems with the API you're using. Cause you don't know that you need to use `useTransition` in the first place. This section helps cross-API discovery.

Reply:
I can add this if it's not super clear what I mean, but I do think it should be here